src/components/tableChildren/AttributeSelect.tsx

```diff
@@ -0,0 +1,114 @@
+import { useState } from 'react';
+import { useApplicationState } from '../../Store';
+import { UiTableRowDatatype } from './TableRow';
+import { useParams } from 'react-router-dom';
+
+type Row = {
+  rowName: string;
+  rowDatatype: UiTableRowDatatype;
+  rowAttributes: string;
+}
+
+type AttributeSelectProps = {
+  row: Row,
+  handleChange: (newValue: string) => void
+}
+
+export default function AttributeSelect({
+  row,
+  handleChange
+}: AttributeSelectProps) {
+  const { id } = useParams();
+  const tables = useApplicationState(state => (
+    state.schemaTables.filter(table => table.id !== id)
+  ));
```

Comment:
This would not support hinting recursive relations.
```suggestion
  const tables = useApplicationState(state => state.schemaTables);
```
This should be enough:
```ts
const tables = useApplicationState(state => state.schemaTables);
```

Reply:
Maybe the selected table name should be passed in as a prop, since it is possible to change the selected table name and create tablerows with recursive FK at the same time.

Or maybe it would be best to keep it as is (filter out the selected table). This would keep the component more independent.

What do you think?

Reply:
Sorry for the late reply. I'm trying to go through your comments one by one.

I'm a FE developer with several years of experience, but I'm not too familiar with databases. I only know as much as I've seen at work or heard on the databases lecture I'm currently taking.

Do you wish to have or to avoid recursive relations? I understand that "recursion" in this context would mean a FK attribute with a table name pointing to the table within which it is. I thought this thing is not allowed, therefore I filtered the active table out. 

I'm not too sure about passing the table name in as a props. I think it's a good idea to keep your component's interface as small as possible, especially if you have already implemented Redux.